pkg/pillar/cmd/client/client.go

```diff
@@ -516,42 +575,98 @@ func selfRegister(zedcloudCtx zedcloud.ZedCloudContext, tlsConfig *tls.Config, d
 		log.Errorf("%s\n", string(contents))
 		done = false
 	}
-	return done
+
+	return done, rtf == types.ExtStatusCertUnknownAuthority
+}
+
+// fetch V2 certs from cloud, return GotCloudCerts and ServerIsV1 boolean
+// if got certs, the leaf is saved to types.ServerCertName file
+func fetchCertChain(zedcloudCtx zedcloud.ZedCloudContext, tlsConfig *tls.Config, retryCount int, force bool) (bool, bool) {
+	var resp *http.Response
+	var b, contents []byte
+	var done bool
+
+	if !force {
+		_, err := os.Stat(types.ServerCertName)
+		if err == nil {
+			return true, false
+		}
+	}
+
+	// certs API is always V2, and without UUID, use http for now
+	requrl := zedcloud.URLPathString(serverNameAndPort, true, true, nilUUID, "certs")
+	// currently there is no data included for the request, same as myGet()
+	done, resp, _, contents = myPost(zedcloudCtx, tlsConfig, requrl, retryCount, int64(len(b)), bytes.NewBuffer(b))
+	if resp != nil {
+		log.Infof("client fetchCertChain done %v, resp-code %d, content len %d", done, resp.StatusCode, len(contents))
+		if resp.StatusCode == http.StatusNotFound || resp.StatusCode == http.StatusUnauthorized ||
```

Comment:
Better to do this as a case statement, and I'd expect to see a case for OK, plus a default case to catch any suprising http status codes. (For instance, we can get GatewayTimeout when things are broken, and I think a different one during a controller update; this code treats those as "OK" which is incorrect.)

Reply:
ok. but still, the last catch is the 'done' is set or not. so the default case if not done, will be false also.

Reply:
OK, but a bit hard to read since done is set in myPost